Intersect.Client/Entities/Player.cs

```diff
@@ -1986,49 +2056,97 @@ private void ProcessDirectionalInput()
                                 OffsetY = 0;
                                 OffsetX = Options.TileWidth;
                             }
 
                             break;
-                        case 3: // Right
+                        case Direction.Right:
                             if (IsTileBlocked(X + 1, Y, Z, MapId, ref blockedBy) == -1)
                             {
                                 tmpX++;
                                 IsMoving = true;
-                                Dir = 3;
+                                Dir = Direction.Right;
                                 OffsetY = 0;
                                 OffsetX = -Options.TileWidth;
                             }
 
+                            break;
+                        case Direction.UpLeft:
+                            if (IsTileBlocked(X - 1, Y - 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY--;
+                                tmpX--;
+                                IsMoving = true;
+                                Dir = Direction.UpLeft;
+                                OffsetY = Options.TileHeight;
+                                OffsetX = Options.TileWidth;
+                            }
+
+                            break;
+                        case Direction.UpRight:
+                            if (IsTileBlocked(X + 1, Y - 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY--;
+                                tmpX++;
+                                IsMoving = true;
+                                Dir = Direction.UpRight;
+                                OffsetY = Options.TileHeight;
+                                OffsetX = -Options.TileWidth;
+                            }
+
+                            break;
+                        case Direction.DownRight:
+                            if (IsTileBlocked(X + 1, Y + 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY++;
+                                tmpX++;
+                                IsMoving = true;
+                                Dir = Direction.DownRight;
+                                OffsetY = -Options.TileHeight;
+                                OffsetX = -Options.TileWidth;
+                            }
+
+                            break;
+                        case Direction.DownLeft:
+                            if (IsTileBlocked(X - 1, Y + 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY++;
+                                tmpX--;
+                                IsMoving = true;
+                                Dir = Direction.DownLeft;
+                                OffsetY = -Options.TileHeight;
+                                OffsetX = Options.TileWidth;
+                            }
+
```

Comment:
Like I mentioned in a previous file, the switch can be set deltas, and then 1 if statement after the switch case can make the assignments/comparisons